docs/resources/mapreduce_cluster.md

```diff
@@ -0,0 +1,354 @@
+  the MapReduce cluster.
+  If using the specified security group, the group need to open the specified port (9022) rules.
+
+* `analysis_core_nodes` - (Optional, List) Specifies a list of the informations about the analysis core nodes in the MapReduce cluster.
+  The `nodes` object structure of the `analysis_core_nodes` is documented below.
+
+* `streaming_core_nodes` - (Optional, List) Specifies a list of the informations about the streaming core nodes in the MapReduce cluster.
+  The `nodes` object structure of the `streaming_core_nodes` is documented below.
+
+* `analysis_task_nodes` - (Optional, List) Specifies a list of the informations about the analysis task nodes in the MapReduce cluster.
+  The `nodes` object structure of the `analysis_task_nodes` is documented below.
+
+* `streaming_task_nodes` - (Optional, List) Specifies a list of the informations about the streaming task nodes in the MapReduce cluster.
+  The `nodes` object structure of the `streaming_task_nodes` is documented below.
+
+* `tags` - (Optional, Map, ForceNew) Specifies the key/value pairs to associate with the cluster.
+
+The `nodes` block supports:
+
+* `flavor` - (Required, String, ForceNew) Specifies the instance specifications for each nodes in node group.
+  Changing this will create a new MapReduce cluster resource.
+
+* `node_number` - (Required, Int) Specifies the number of nodes for the node group.
+
+  -> **NOTE:** Only the core group and task group updations are allowed.
+
+  The number of nodes after scaling cannot be less than the number of nodes originally created.
+
+* `root_volume_type` - (Required, String, ForceNew) Specifies the system disk flavor of the nodes.
+  Changing this will create a new MapReduce cluster resource.
+
+* `root_volume_size` - (Required, Int, ForceNew) Specifies the system disk size of the nodes.
+  Changing this will create a new MapReduce cluster resource.
+
+* `data_volume_type` - (Optional, String, ForceNew) Specifies the data disk flavor of the nodes.
+  Required if `data_volume_count` is not empty.
+  Changing this will create a new MapReduce cluster resource.
+
+* `data_volume_size` - (Optional, Int, ForceNew) Specifies the data disk size of the nodes.
+  Required if `data_volume_count` is not empty.
+  Changing this will create a new MapReduce cluster resource.
+
+* `data_volume_count` - (Optional, Int, ForceNew) Specifies the data disk number of the nodes.
+  Changing this will create a new MapReduce cluster resource.
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The cluster ID in UUID format.
+* `total_node_number` - The total number of nodes deployed in a cluster.
+* `master_node_ip` - The IP address of a master node.
+* `external_ip` - The external IP address.
+* `private_ip` - The preferred private IP address.
+* `internal_ip` - The internal IP address.
+* `external_alternate_ip` - IP address of a Master node.
```

Comment:
what's the differents between xx_ip ?

Reply:
The `private_ip` is the internal ip of the HA master node of the master node group.